### DISCUSSION TITLE
Disable dev mode message in web test runner

### DISCUSSION THREAD

**Discussion by bendera** [lit/lit#4817] [Q&A] [answered]
Hi,

Is there any way to disable the dev mode messages in the web test runner? This is the message I'm referring to:
```
 Lit is in dev mode. Not recommended for production! See https://lit.dev/msg/dev-mode for more information.
 ```
 I have around 30 components in my library and every component writes this message to the console. I read the documentation carefully, and if I understand correctly, I need to set the exportConditions in the test runner configuration, something like this:
 ```
 import { nodeResolve } from '@rollup/plugin-node-resolve';

export default {
  // ...
  plugins: [nodeResolve({
    exportConditions: ['default']
  })]
};
 ```
 It doesn't work. Did I miss something?

**Comment by justinfagnani** [comment, marked as answer]
We try to deduplicate the warning by recording if we logged it already in a global. Are you running tests in separate globals?

You can filter logs in WTR with the `filterBrowserLogs` option:

```ts
  // Only log actual errors
  filterBrowserLogs: ({type}) => type === 'error',
```

or maybe:
```ts
  filterBrowserLogs: ({args}) => !args.startsWith('Lit is in dev mode'),
```

As for export conditions, I'm not 100% clear on if when requesting files you want the "default" condition, or the "import" condition. Did you try "import"?

Of course, the problem is that you won't get Lit's extra dev mode logging which can be helpful for more expensive-at-runtime to detect problems.

**Comment by bendera** [comment]
> Are you running tests in separate globals?

I don't know. I think WTR runs every test file separately. Only one message is logged/file.

> As for export conditions, I'm not 100% clear on if when requesting files you want the "default" condition, or the "import" condition. Did you try "import"?

I'm not 100% sure what I'm doing. :)  I assumed that if I set 'exportConditions' to default, the Node resolver plugin would import files from the package's root directory instead of the 'development' directory. 

> Of course, the problem is that you won't get Lit's extra dev mode logging which can be helpful for more expensive-at-runtime to detect problems.

It's clear. In the browser, it logs only once, so it's not annoying there.

> You can filter logs in WTR with the filterBrowserLogs option

That's what I'm looking for, thank you!! I skimmed the docs, I thought the cli parameters and config options are the same.
